hpat/hiframes_typed.py

```diff
@@ -751,6 +754,96 @@ def f(arr, on_arr, w, center):  # pragma: no cover
             return nodes
         return [assign]
 
+    def _handle_series_combine(self, assign, lhs, rhs, series_var):
+        """translate s1.combine(s2,lambda x1,x2 :...) to prange()
+        """
+        func = guard(get_definition, self.func_ir, rhs.args[1])
+        if func is None or not (isinstance(func, ir.Expr)
+                                and func.op == 'make_function'):
+            raise ValueError("lambda for combine not found")
+
+        out_typ = self.typemap[lhs.name].dtype
+
+        # If we are called with 3 arguments, we must use 3rd arg as a fill value,
+        # instead of Nan.
+        use_nan = len(rhs.args) == 2
+
+        # prange func to inline
+        if use_nan:
+            func_text = "def f(A, B):\n"
+        else:
+            func_text = "def f(A, B, C):\n"
+        func_text += "  numba.parfor.init_prange()\n"
+        func_text += "  n1 = len(A)\n"
+        func_text += "  n2 = len(B)\n"
+        func_text += "  n = n1\n"
+        func_text += "  if n2 > n:\n"
```

Comment:
Control flow currently prevents the `init_prange` optimizations. Using `max(n1, n2)` is better here.